jsonrpc/errors.go

```diff
@@ -90,3 +92,12 @@ func NewInternalError(msg string) *internalError {
 func NewSubscriptionNotFoundError(method string) *subscriptionNotFoundError {
 	return &subscriptionNotFoundError{fmt.Sprintf("subscribe method %s not found", method)}
 }
+
+func constructErrorFromRevert(result *runtime.ExecutionResult) error {
+	revertErrMsg, unpackErr := abi.UnpackRevertError(result.ReturnValue)
+	if unpackErr != nil {
+		return result.Err
```

Comment:
Just for clarification, you are calling this method only if the error is revert. If you get unpackErr what does that mean?

Reply:
It's just [basic error handling](https://github.com/umbracle/ethgo/blob/50195055efb0377415550beca8c97b892edefdc8/abi/revert.go#L10-L23) from the `go-web3` package, this error doesn't have to be a string error, but can be bundled regardless